src/Compilers/Core/Portable/DiagnosticAnalyzer/UnresolvedAnalyzerReference.cs

```diff
@@ -36,9 +36,12 @@ public override string FullPath
             }
         }
 
-        public override bool IsUnresolved
+        public override string Id
         {
-            get { return true; }
+            get
+            {
+                return _unresolvedPath;
```

Comment:
I've added a check in the constructor that the path is not null--if you don't have a path at all, you shouldn't be using `UnresolvedAnalyzerReference`.